sycl/test-e2e/USM/P2P/p2p_atomics.cpp

```diff
@@ -51,26 +42,26 @@ int main() {
   // Enables Devs[1] to access Devs[0] memory.
   Devs[1].ext_oneapi_enable_peer_access(Devs[0]);
 
-  std::vector<double> input(N);
+  std::vector<int> input(N);
   std::iota(input.begin(), input.end(), 0);
 
-  double h_sum = 0.;
+  int h_sum = 0.;
   for (const auto &value : input) {
     h_sum += value;
   }
 
-  double *d_sum = malloc_shared<double>(1, Queues[0]);
-  double *d_in = malloc_device<double>(N, Queues[0]);
+  int *d_sum = malloc_shared<int>(1, Queues[0]);
```

Comment:
Is P2P allowed for Shared USM allocations? The extensions says https://github.com/intel/llvm/blob/e374c69ffc2f24f583fcc26b1628b4daff8fdb65/sycl/doc/extensions/supported/sycl_ext_oneapi_peer_access.asciidoc?plain=1#L57-L58 so either this code is wrong or the extension wasn't updated.

Reply:
Also, the extension says https://github.com/intel/llvm/blob/e374c69ffc2f24f583fcc26b1628b4daff8fdb65/sycl/doc/extensions/supported/sycl_ext_oneapi_peer_access.asciidoc?plain=1#L89-L90, yet we use `sycl::queue{D}` on line 32. Are we relying on default context here?

Reply:
> Is P2P allowed for Shared USM allocations? The extensions says
> 
> https://github.com/intel/llvm/blob/e374c69ffc2f24f583fcc26b1628b4daff8fdb65/sycl/doc/extensions/supported/sycl_ext_oneapi_peer_access.asciidoc?plain=1#L57-L58
> so either this code is wrong or the extension wasn't updated.

You're right, I should have used device memory. shared memory works in the circumstance for nvidia and amd, but like you point out the extension doesn't support it. I imagine this might cause flaky failures on l0 if it doesn't support it.

Jack

Reply:
> Also, the extension says
> 
> https://github.com/intel/llvm/blob/e374c69ffc2f24f583fcc26b1628b4daff8fdb65/sycl/doc/extensions/supported/sycl_ext_oneapi_peer_access.asciidoc?plain=1#L89-L90
> , yet we use `sycl::queue{D}` on line 32. Are we relying on default context here?

Yeah this test relies on the default context and in this sense I believe it is valid code?
Also I think that in real world codes the default context is used virtually always.